common/src/main/java/com/graphicsfuzz/common/util/GloballyTruncateLoops.java

```diff
@@ -39,9 +64,90 @@
    */
   public static void truncate(ShaderJob shaderJob, int loopLimit, String loopCountName,
                               String loopBoundName) {
-    // TODO(https://github.com/google/graphicsfuzz/issues/866): Implement loop limiting
-    //  logic.
-    throw new RuntimeException("Not implemented yet.");
-  }
+    for (TranslationUnit tu : shaderJob.getShaders()) {
+      Declaration firstNonPrecisionDeclaration = null;
+      for (Declaration decl : tu.getTopLevelDeclarations()) {
+        if (decl instanceof PrecisionDeclaration) {
+          continue;
+        }
+        firstNonPrecisionDeclaration = decl;
+        break;
+      }
```

Comment:
Assert that firstNonPrecisionDeclaration is not null at this point, as we should always find something.

Reply:
done